templates/demo-store/app/hooks/usePageAnalytics.tsx

```diff
@@ -0,0 +1,33 @@
+import {useMatches} from '@remix-run/react';
+import type {ShopifyPageViewPayload} from '@shopify/hydrogen';
+import {useMemo} from 'react';
+
+import {DEFAULT_LOCALE} from '~/lib/utils';
+
+export function usePageAnalytics({hasUserConsent}: {hasUserConsent: boolean}) {
+  const matches = useMatches();
+
+  const analyticsFromMatches = useMemo(() => {
+    const data: Record<string, unknown> = {};
+
+    matches.forEach((event) => {
+      const eventData = event?.data;
+      if (eventData) {
+        eventData['analytics'] && Object.assign(data, eventData['analytics']);
+
+        const selectedLocale = eventData['selectedLocale'] || DEFAULT_LOCALE;
+        Object.assign(data, {
+          currency: selectedLocale.currency,
+          acceptedLanguage: selectedLocale.language,
+        });
+      }
+    });
+
+    return data;
+  }, [matches]);
+
+  return {
+    ...analyticsFromMatches,
+    hasUserConsent,
+  } as unknown as ShopifyPageViewPayload;
```

Comment:
Nit picky, but if this return  is moved to be within the `useMemo`, then a new object won't be created on every render with the `...` destructuring.